src/Component/Doctrine/Common/State/PersistProcessor.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Resource\Doctrine\Common\State;
+
+use Doctrine\Persistence\ManagerRegistry;
+use Sylius\Component\Resource\Context\Context;
+use Sylius\Component\Resource\Context\Option\MetadataOption;
+use Sylius\Component\Resource\Metadata\Operation;
+use Sylius\Component\Resource\State\ProcessorInterface;
+
+final class PersistProcessor implements ProcessorInterface
+{
+    public function __construct(private ManagerRegistry $managerRegistry)
+    {
+    }
+
+    public function process(mixed $data, Operation $operation, Context $context): mixed
+    {
+        $metadata = $context->get(MetadataOption::class)?->metadata();
```

Comment:
I'd like to retrieve a ResourceOption::class instead.
I'd like to get rid of the current MetadataInterface and using the new Resource metadata instead.